src/napari/layers/utils/stack_utils.py

```diff
@@ -334,13 +336,38 @@ def images_to_stack(images: list[Image], axis: int = 0, **kwargs) -> Image:
                     n_scales_list=n_scales_list,
                 )
             )
-        n_scales = n_scales_list[0]
+        arrays_to_check = [img.data[0] for img in images]
+    else:
+        arrays_to_check = [img.data for img in images]
+
+    # check if any of the data arrays are zarr arrays
+    # zarr doesn't have a stack method, so we will use dask.array.stack
+    is_zarr = any(
+        hasattr(arr, '__module__') and arr.__module__.startswith('zarr')
+        for arr in arrays_to_check
+    )
```

Comment:
WE should have some `is_zarr`, `is_tensorstore` etc methods in some utils module to not replicate such patterns across the codebase.

Reply:
If you open an issue with the task, I'm ok with that.